include/Hardmon.hpp

```diff
@@ -0,0 +1,230 @@
+    /** Motor Controller Toggle Override Pin */
+    static constexpr IO::Pin MOTOR_CONTROLLER_TOGGLE_OVERRIDE_PIN = IO::Pin::PC_8;
+
+    /** Accessory CAN RX Pin */
+    static constexpr IO::Pin ACCESSORY_CAN_RX_PIN = IO::Pin::PA_11;
+    /** Accessory CAN TX Pin */
+    static constexpr IO::Pin ACCESSORY_CAN_TX_PIN = IO::Pin::PA_12;
+
+    /** Powertrain Network CAN RX Pin */
+    static constexpr IO::Pin POWERTRAIN_CAN_RX_PIN = IO::Pin::PB_12;
+    /** Powertrain Network CAN TX Pin */
+    static constexpr IO::Pin POWERTRAIN_CAN_TX_PIN = IO::Pin::PB_13;
+
+    /** Watchdog Pin */
+    static constexpr IO::Pin WATCHDOG_PIN = IO::Pin::PB_4;
+    /** Hardmon Fault Pin */
+    static constexpr IO::Pin HM_FAULT_PIN = IO::Pin::PB_5;
+
+    /** Ignition "12v" Pin */
+    static constexpr IO::Pin IGNITION_CHECK_PIN = IO::Pin::PB_6;
+    /** Ignition 3V3 Pin */
+    static constexpr IO::Pin IGNITION_3V3_PIN = IO::Pin::PC_12;
+
+    /** Estop "12v" Pin */
+    static constexpr IO::Pin ESTOP_CHECK_PIN = IO::Pin::PB_7;
+    /** Estop 3V3 Pin */
+    static constexpr IO::Pin ESTOP_3V3_PIN = IO::Pin::PD_2;
+
+    /** MicroController Reset Pin */
+    static constexpr IO::Pin UC_RESET_PIN = IO::Pin::PB_8;
+    /** CAN Selftest Pin */
+    static constexpr IO::Pin CAN_SELFTEST_PIN = IO::Pin::PB_9;
+    /** MCUC CAN Override Pin */
+    static constexpr IO::Pin CAN_OVERRIDE_PIN = IO::Pin::PC_13;
+
+    /**
+     * Constructor for Hardmon object
+     */
+    Hardmon(reqGPIO gpios, IO::CAN& ptCAN);
+
+    /**
+     * Get a pointer to the start of the object dictionary
+     *
+     * @return Pointer to the start of the object dictionary
+     */
+    CO_OBJ_T* getObjectDictionary() override;
+
+    /**
+     * Get the number of elements in the object dictionary.
+     *
+     * @return The number of elements in the object dictionary
+     */
+    uint8_t getNumElements() override;
+
+    /**
+     * Handles the passed in Powertrain CAN message.
+     *
+     * @param message message to handle
+     */
+    void handlePowertrainCanMessage(IO::CANMessage& message);
+
+    /**
+    * Get the device's node ID
+    *
+    * @return The node ID of the can device.
+     */
+    uint8_t getNodeID() override;
+
+    /**
+     * Returns a pointer to the queue for Powertrain CANopen messages
+     * @return
+     */
+    EVT::core::types::FixedQueue<POWERTRAIN_QUEUE_SIZE, IO::CANMessage>* getPowertrainQueue();
+
+    void process();
+
+private:
+    /**
+     * Local instance of PowertrainCan (handles PowertrainCAN messages
+     */
+    DEV::PowertrainCAN powertrainCAN;
+
+    /**
+     * The model that is determining our control flow
+     * Automatically constructed here (not to be passed in)
+     */
+    Hardmon_Model model;
+
+    //TODO: ask EEs about initial values (i.e. if they should be 0 or whatever)
+
+    //Model input data
+
+    bool forwardEnable;///< CAN (HIB): handlebar forward enable
```

Comment:
This is our standard for documenting this kind of thing
```suggestion
    /** CAN (HIB): handlebar forward enable */
    bool forwardEnable;
```